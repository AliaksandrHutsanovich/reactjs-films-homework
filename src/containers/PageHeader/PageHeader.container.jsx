import React, { memo, useContext } from 'react';
import PropTypes from 'prop-types';
import { BackgroundContext } from '../../contexts';
import PageHeader from '../../pages/FilmsPage/PageHeader';

const PageHeaderContainer = ({ onAction, withBackground }) => {
  const { img } = useContext(BackgroundContext);

  return (
    <PageHeader
      onAction={onAction}
      withBackground={withBackground}
      imgUrl={img}
    />
  );
};

PageHeaderContainer.propTypes = {
  onAction: PropTypes.func.isRequired,
  withBackground: PropTypes.bool,
};

PageHeaderContainer.defaultProps = {
  withBackground: null,
};

export default memo(PageHeaderContainer);
