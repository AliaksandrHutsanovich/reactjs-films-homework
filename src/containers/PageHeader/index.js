export { default } from './PageHeader.container';
