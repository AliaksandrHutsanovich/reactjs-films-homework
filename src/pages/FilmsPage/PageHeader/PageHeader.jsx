import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Layout } from 'antd';
import {
  SearchInput,
  HeadLine,
} from '../../../components';

import styles from '../FilmsPage.scss';

const { Header } = Layout;

const PageHeader = ({ onAction }) => (
  <Header className={styles.header}>
    <HeadLine text="FILMS" level={3} fontFamily="arial" />
    <SearchInput onSearch={onAction} />
  </Header>
);

PageHeader.propTypes = {
  onAction: PropTypes.func.isRequired,
};

export default memo(PageHeader);
