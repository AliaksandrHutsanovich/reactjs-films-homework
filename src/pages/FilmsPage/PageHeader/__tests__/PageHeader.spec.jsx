import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PageHeader from '../PageHeader';

describe('<PageHeader />', () => {
  const renderer = new ShallowRenderer();

  it('should be rendered properly', () => {
    renderer.render(<PageHeader onAction={() => {}} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
