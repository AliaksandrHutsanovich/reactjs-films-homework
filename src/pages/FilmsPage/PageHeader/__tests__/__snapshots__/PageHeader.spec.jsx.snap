// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> should be rendered properly 1`] = `
<Header
  className="header"
>
  <UNDEFINED
    fontFamily="arial"
    level={3}
    text="FILMS"
  />
  <UNDEFINED
    onSearch={[Function]}
  />
</Header>
`;
